compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -397,7 +401,7 @@ impl<D: Deps> DepGraphData<D> {
 
     /// Executes something within an "anonymous" task, that is, a task the
     /// `DepNode` of which is determined by the list of inputs it read from.
```

Comment:
Please add a comment that this does not actually read the dep_node and has to be done manually?

Or even better, change the return type to use `WithDepNode<T>/`WithOptDepNode<T>`. Or at least add a FIXME for that here